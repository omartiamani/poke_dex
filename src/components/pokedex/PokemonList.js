import React, { Component } from "react";
import { observer } from "mobx-react";
import { withStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";

import PokemonCard from "./PokemonCard";

const styles = theme => ({
  root: {
    display: "flex",
    flexDirection: "row"
  },
  pokemonCard: {
    width: "100%"
  }
});

@observer //We make the component oserver to update if the observable data in the store change
/**
 * The PokemonList component class.
 * Display the list of cards of all created pokemons
 */
class PokemonList extends Component {
  render() {
    const { classes, pokemonList } = this.props;

    //Transform the list of pokemon object into a list of Card component
    const PokemonListComponent = pokemonList.map(pokemon => (
      <Grid key={pokemon.id} item xs={3}>
        <PokemonCard
          className={classes.pokemonCard}
          id={pokemon.id}
          name={pokemon.name}
          type={pokemon.type}
          description={pokemon.description}
        />
      </Grid>
    ));

    //Display
    return (
      <Grid container spacing={3}>
        {PokemonListComponent}
      </Grid>
    );
  }
}

export default withStyles(styles)(PokemonList);
