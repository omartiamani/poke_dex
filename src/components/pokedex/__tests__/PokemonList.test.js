import React from "react";
import PokemonList from "../PokemonList";
import { shallow } from "enzyme";

it("renders correctly", () => {
  const pokemonList = [
    {
      id: 0,
      name: "Pikachu",
      type: "Foudre",
      description: "Pokemon au teint jaune"
    },
    {
      id: 1,
      name: "Bulbizard",
      type: "Plante",
      description: "Pokemon au teint vert"
    },
    {
      id: 2,
      name: "Salameche",
      type: "Feu",
      description: "Pokemon au teint rouge"
    }
  ];
  const component = shallow(<PokemonList pokemonList={pokemonList} />);

  expect(component).toMatchSnapshot();
});
