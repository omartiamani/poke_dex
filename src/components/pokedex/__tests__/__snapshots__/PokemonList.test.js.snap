// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<PokemonList
  classes={
    Object {
      "pokemonCard": "PokemonList-pokemonCard-2",
      "root": "PokemonList-root-1",
    }
  }
  pokemonList={
    Array [
      Object {
        "description": "Pokemon au teint jaune",
        "id": 0,
        "name": "Pikachu",
        "type": "Foudre",
      },
      Object {
        "description": "Pokemon au teint vert",
        "id": 1,
        "name": "Bulbizard",
        "type": "Plante",
      },
      Object {
        "description": "Pokemon au teint rouge",
        "id": 2,
        "name": "Salameche",
        "type": "Feu",
      },
    ]
  }
/>
`;
